arm-network/2016-09-01/swagger/routeFilter.json

```diff
@@ -0,0 +1,849 @@
+      "get": {
+        "tags": [
+          "ServiceCommunities"
+        ],
+        "operationId": "ServiceCommunities_List",
+        "description": "Gets all the available service communities.",
+        "parameters": [
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "#/parameters/SubscriptionIdParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Request successful. The operation returns a list of ServiceCommunity resources.",
+            "schema": {
+              "$ref": "#/definitions/ServiceCommunityListResult"
+            }
+          }
+        },
+        "x-ms-pageable": {
+          "nextLinkName": "nextLink"
+        }
+      }
+    }
+  },
+  "definitions": {
+    "RouteFilterRulePropertiesFormat": {
+      "properties": {
+        "access": {
+          "type": "string",
+          "description": "The access type of the rule. Possible values are: 'Allow', 'Deny",
+          "enum": [
+            "Allow",
+            "Deny"
+          ],
+          "x-ms-enum": {
+            "name": "Access",
+            "modelAsString": true
+          }
+        },
+        "routeFilterRuleType": {
+          "type": "string",
+          "description": "The type of the rule. Possible values are: 'Community",
+          "enum": [
+            "Community"
+          ],
+          "x-ms-enum": {
+            "name": "RouteFilterRuleType",
+            "modelAsString": true
+          }
+        },
+        "communities": {
+          "type": "array",
+          "items": {
+            "type": "string"
+          },
+          "description": "The collection for bgp communities to filter on."
+        },
+        "provisioningState": {
+          "type": "string",
+          "description": "The provisioning state of the resource. Possible values are: 'Updating', 'Deleting', 'Succeeded' and 'Failed'."
+        }
+      },
+      "required": [
+        "nextHopType"
+      ],
+      "description": "Route Filter Rule Resource"
+    },
+    "RouteFitlerRule": {
+      "properties": {
+        "properties": {
+          "x-ms-client-flatten": true,
+          "$ref": "#/definitions/RouteFilterRulePropertiesFormat"
+        },
+        "name": {
+          "type": "string",
+          "description": "The name of the resource that is unique within a resource group. This name can be used to access the resource."
+        },
+        "etag": {
+          "type": "string",
+          "description": "A unique read-only string that changes whenever the resource is updated."
+        },
+        "type": {
+          "readOnly": true,
+          "type": "string",
+          "description": "Resource type."
+        },
+        "tags": {
+          "type": "object",
+          "additionalProperties": {
+            "type": "string"
+          },
+          "description": "Resource tags."
+        }
+      },
+      "allOf": [
```

Comment:
To me RouteFilterRule is not a "SubResource". It is a "ChildResource". A SubResource only has an id which is settable. Whereas a ChildResource is very much like a Resource. Most of the times from ARM perspective, it is not a TrackedResource. This is the primary difference and hence has an effect on the "requiredness" of "location" property. For a TrackedResource "location" is required whereas for a ChildResource location is not required.

This model has name, type, tags (properties) similar to a Resource object. So how can this be a SubResource? A SubResource is a simple reference to the resource.

Is the id property of RouteFilterRule settable ?

/cc @anudeepsharma , @fearthecowboy , @johanste, @ravbhatnagar - This is an example, of why we need common definitions. The earlier we get them from the ARM team, better it is for all the RPs.